// ngx-bootstrap/alert/alert.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ngx-bootstrap/alert/alert.component';
import * as i3 from 'ngx-bootstrap/alert/alert.config';
var styles_AlertComponent = [];
export var RenderType_AlertComponent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_AlertComponent, data: {} });
function View_AlertComponent_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(1, 0, null, null, 7, 'button', [['aria-label', 'Close'], ['class', 'close'],
            ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.close() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])),
        (_l()(), i0.ɵeld(3, 0, null, null, 1, 'span', [['aria-hidden', 'true']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['×'])), (_l()(), i0.ɵted(-1, null, ['\n        '])),
        (_l()(), i0.ɵeld(6, 0, null, null, 1, 'span', [['class', 'sr-only']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['Close'])), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵted(-1, null, ['\n    ']))], null, null);
}
function View_AlertComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(), i0.ɵeld(1, 0, null, null, 7, 'div', [['role', 'alert']], null, null, null, null, null)), i0.ɵdid(2, 278528, null, 0, i1.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1, 'ngClass'] }, null), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_AlertComponent_2)), i0.ɵdid(5, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(),
            i0.ɵted(-1, null, ['\n    '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(), i0.ɵted(-1, null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = ('alert alert-' + _co.type);
        var currVal_1 = _co.classes;
        _ck(_v, 2, 0, currVal_0, currVal_1);
        var currVal_2 = _co.dismissible;
        _ck(_v, 5, 0, currVal_2);
    }, null);
}
export function View_AlertComponent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_AlertComponent_1)), i0.ɵdid(2, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = !_co.isClosed;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
export function View_AlertComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'alert', [], null, null, null, View_AlertComponent_0, RenderType_AlertComponent)),
        i0.ɵdid(1, 114688, null, 0, i2.AlertComponent, [i3.AlertConfig], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
export var AlertComponentNgFactory = i0.ɵccf('alert,ngx-alert', i2.AlertComponent, View_AlertComponent_Host_0, { type: 'type', dismissible: 'dismissible',
    dismissOnTimeout: 'dismissOnTimeout' }, { onClose: 'onClose', onClosed: 'onClosed' }, ['*']);
