// ngx-bootstrap/popover/popover-container.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ngx-bootstrap/popover/popover-container.component';
import * as i3 from 'ngx-bootstrap/popover/popover.config';
var styles_PopoverContainerComponent = ['.bs-popover-top[_nghost-%COMP%]   .arrow[_ngcontent-%COMP%], .bs-popover-bottom[_nghost-%COMP%]   .arrow[_ngcontent-%COMP%] {\n      left: calc(50% - 5px);\n    }\n    .bs-popover-left[_nghost-%COMP%]   .arrow[_ngcontent-%COMP%], .bs-popover-right[_nghost-%COMP%]   .arrow[_ngcontent-%COMP%] {\n      top: calc(50% - 2.5px);\n    }'];
export var RenderType_PopoverContainerComponent = i0.ɵcrt({ encapsulation: 0,
    styles: styles_PopoverContainerComponent, data: {} });
function View_PopoverContainerComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'h3', [['class',
                'popover-title popover-header']], null, null, null, null, null)), (_l()(), i0.ɵted(1, null, ['', '']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.title;
        _ck(_v, 1, 0, currVal_0);
    });
}
export function View_PopoverContainerComponent_0(_l) {
    return i0.ɵvid(2, [(_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵeld(1, 0, null, null, 0, 'div', [['class', 'popover-arrow arrow']], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_PopoverContainerComponent_1)), i0.ɵdid(4, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵeld(5, 0, null, null, 1, 'div', [['class', 'popover-content popover-body']], null, null, null, null, null)),
        i0.ɵncd(null, 0), (_l()(), i0.ɵted(-1, null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.title;
        _ck(_v, 4, 0, currVal_0);
    }, null);
}
export function View_PopoverContainerComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'popover-container', [['role', 'tooltip'], ['style', 'display:block;']], [[8, 'className', 0], [2, 'show',
                null]], null, null, View_PopoverContainerComponent_0, RenderType_PopoverContainerComponent)), i0.ɵdid(1, 49152, null, 0, i2.PopoverContainerComponent, [i3.PopoverConfig], null, null)], null, function (_ck, _v) {
        var currVal_0 = (((((((('popover in popover-' + i0.ɵnov(_v, 1).placement) + ' ') + 'bs-popover-') + i0.ɵnov(_v, 1).placement) + ' ') + i0.ɵnov(_v, 1).placement) + ' ') + i0.ɵnov(_v, 1).containerClass);
        var currVal_1 = !i0.ɵnov(_v, 1).isBs3;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
export var PopoverContainerComponentNgFactory = i0.ɵccf('popover-container', i2.PopoverContainerComponent, View_PopoverContainerComponent_Host_0, { placement: 'placement',
    title: 'title' }, {}, ['*']);
