// ngx-bootstrap/datepicker/datepicker-inner.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ngx-bootstrap/datepicker/datepicker-inner.component';
var styles_DatePickerInnerComponent = [];
export var RenderType_DatePickerInnerComponent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_DatePickerInnerComponent, data: {} });
function View_DatePickerInnerComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 3, 'div', [['class',
                'well well-sm bg-faded p-a card'], ['role', 'application']], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n      '])), i0.ɵncd(null, 0), (_l()(), i0.ɵted(-1, null, ['\n    ']))], null, null);
}
export function View_DatePickerInnerComponent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DatePickerInnerComponent_1)),
        i0.ɵdid(2, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.datepickerMode;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
export function View_DatePickerInnerComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'datepicker-inner', [], null, null, null, View_DatePickerInnerComponent_0, RenderType_DatePickerInnerComponent)), i0.ɵdid(1, 638976, null, 0, i2.DatePickerInnerComponent, [], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
export var DatePickerInnerComponentNgFactory = i0.ɵccf('datepicker-inner', i2.DatePickerInnerComponent, View_DatePickerInnerComponent_Host_0, { locale: 'locale',
    datepickerMode: 'datepickerMode', startingDay: 'startingDay', yearRange: 'yearRange',
    minDate: 'minDate', maxDate: 'maxDate', minMode: 'minMode', maxMode: 'maxMode', showWeeks: 'showWeeks',
    formatDay: 'formatDay', formatMonth: 'formatMonth', formatYear: 'formatYear', formatDayHeader: 'formatDayHeader',
    formatDayTitle: 'formatDayTitle', formatMonthTitle: 'formatMonthTitle', onlyCurrentMonth: 'onlyCurrentMonth',
    shortcutPropagation: 'shortcutPropagation', customClass: 'customClass', monthColLimit: 'monthColLimit',
    yearColLimit: 'yearColLimit', dateDisabled: 'dateDisabled', initDate: 'initDate',
    activeDate: 'activeDate' }, { selectionDone: 'selectionDone', update: 'update', activeDateChange: 'activeDateChange' }, ['*']);
