// ngx-bootstrap/datepicker/datepicker.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from 'ngx-bootstrap/datepicker/datepicker.component';
import * as i2 from './datepicker-inner.component.ngfactory';
import * as i3 from 'ngx-bootstrap/datepicker/datepicker-inner.component';
import * as i4 from './daypicker.component.ngfactory';
import * as i5 from 'ngx-bootstrap/datepicker/daypicker.component';
import * as i6 from './monthpicker.component.ngfactory';
import * as i7 from 'ngx-bootstrap/datepicker/monthpicker.component';
import * as i8 from './yearpicker.component.ngfactory';
import * as i9 from 'ngx-bootstrap/datepicker/yearpicker.component';
import * as i10 from '@angular/forms';
import * as i11 from 'ngx-bootstrap/datepicker/datepicker.config';
var styles_DatePickerComponent = [];
export var RenderType_DatePickerComponent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_DatePickerComponent, data: {} });
export function View_DatePickerComponent_0(_l) {
    return i0.ɵvid(0, [i0.ɵqud(402653184, 1, { _datePicker: 0 }), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵeld(2, 0, null, null, 11, 'datepicker-inner', [], null, [[null, 'update'], [null, 'selectionDone'],
            [null, 'activeDateChange']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('update' === en)) {
                var pd_0 = (_co.onUpdate($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('selectionDone' === en)) {
                var pd_1 = (_co.onSelectionDone($event) !== false);
                ad = (pd_1 && ad);
            }
            if (('activeDateChange' === en)) {
                var pd_2 = (_co.onActiveDateChange($event) !== false);
                ad = (pd_2 && ad);
            }
            return ad;
        }, i2.View_DatePickerInnerComponent_0, i2.RenderType_DatePickerInnerComponent)),
        i0.ɵdid(3, 638976, [[1, 4]], 0, i3.DatePickerInnerComponent, [], { locale: [0,
                'locale'], datepickerMode: [1, 'datepickerMode'], startingDay: [2, 'startingDay'],
            yearRange: [3, 'yearRange'], minDate: [4, 'minDate'], maxDate: [5, 'maxDate'], minMode: [6,
                'minMode'], maxMode: [7, 'maxMode'], showWeeks: [8, 'showWeeks'], formatDay: [9,
                'formatDay'], formatMonth: [10, 'formatMonth'], formatYear: [11, 'formatYear'],
            formatDayHeader: [12, 'formatDayHeader'], formatDayTitle: [13, 'formatDayTitle'],
            formatMonthTitle: [14, 'formatMonthTitle'], onlyCurrentMonth: [15, 'onlyCurrentMonth'],
            shortcutPropagation: [16, 'shortcutPropagation'], customClass: [17, 'customClass'],
            monthColLimit: [18, 'monthColLimit'], yearColLimit: [19, 'yearColLimit'], dateDisabled: [20,
                'dateDisabled'], initDate: [21, 'initDate'], activeDate: [22, 'activeDate'] }, { selectionDone: 'selectionDone', update: 'update', activeDateChange: 'activeDateChange' }),
        (_l()(), i0.ɵted(-1, 0, ['\n      '])), (_l()(), i0.ɵeld(5, 0, null, 0, 1, 'daypicker', [['tabindex', '0']], null, null, null, i4.View_DayPickerComponent_0, i4.RenderType_DayPickerComponent)), i0.ɵdid(6, 114688, null, 0, i5.DayPickerComponent, [i3.DatePickerInnerComponent], null, null), (_l()(), i0.ɵted(-1, 0, ['\n      '])), (_l()(), i0.ɵeld(8, 0, null, 0, 1, 'monthpicker', [['tabindex',
                '0']], null, null, null, i6.View_MonthPickerComponent_0, i6.RenderType_MonthPickerComponent)), i0.ɵdid(9, 114688, null, 0, i7.MonthPickerComponent, [i3.DatePickerInnerComponent], null, null), (_l()(), i0.ɵted(-1, 0, ['\n      '])), (_l()(), i0.ɵeld(11, 0, null, 0, 1, 'yearpicker', [['tabindex',
                '0']], null, null, null, i8.View_YearPickerComponent_0, i8.RenderType_YearPickerComponent)), i0.ɵdid(12, 114688, null, 0, i9.YearPickerComponent, [i3.DatePickerInnerComponent], null, null), (_l()(), i0.ɵted(-1, 0, ['\n    '])), (_l()(), i0.ɵted(-1, null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.config.locale;
        var currVal_1 = _co.datepickerMode;
        var currVal_2 = _co.startingDay;
        var currVal_3 = _co.yearRange;
        var currVal_4 = _co.minDate;
        var currVal_5 = _co.maxDate;
        var currVal_6 = _co.minMode;
        var currVal_7 = _co.maxMode;
        var currVal_8 = _co.showWeeks;
        var currVal_9 = _co.formatDay;
        var currVal_10 = _co.formatMonth;
        var currVal_11 = _co.formatYear;
        var currVal_12 = _co.formatDayHeader;
        var currVal_13 = _co.formatDayTitle;
        var currVal_14 = _co.formatMonthTitle;
        var currVal_15 = _co.onlyCurrentMonth;
        var currVal_16 = _co.shortcutPropagation;
        var currVal_17 = _co.customClass;
        var currVal_18 = _co.monthColLimit;
        var currVal_19 = _co.yearColLimit;
        var currVal_20 = _co.dateDisabled;
        var currVal_21 = _co.initDate;
        var currVal_22 = _co.activeDate;
        _ck(_v, 3, 1, [currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6,
            currVal_7, currVal_8, currVal_9, currVal_10, currVal_11, currVal_12, currVal_13,
            currVal_14, currVal_15, currVal_16, currVal_17, currVal_18, currVal_19, currVal_20,
            currVal_21, currVal_22]);
        _ck(_v, 6, 0);
        _ck(_v, 9, 0);
        _ck(_v, 12, 0);
    }, null);
}
export function View_DatePickerComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 2, 'datepicker', [], null, null, null, View_DatePickerComponent_0, RenderType_DatePickerComponent)), i0.ɵprd(5120, null, i10.NG_VALUE_ACCESSOR, function (p0_0) {
            return [p0_0];
        }, [i1.DatePickerComponent]), i0.ɵdid(2, 49152, null, 0, i1.DatePickerComponent, [i11.DatepickerConfig], null, null)], null, null);
}
export var DatePickerComponentNgFactory = i0.ɵccf('datepicker', i1.DatePickerComponent, View_DatePickerComponent_Host_0, { datepickerMode: 'datepickerMode',
    initDate: 'initDate', minDate: 'minDate', maxDate: 'maxDate', minMode: 'minMode',
    maxMode: 'maxMode', showWeeks: 'showWeeks', formatDay: 'formatDay', formatMonth: 'formatMonth',
    formatYear: 'formatYear', formatDayHeader: 'formatDayHeader', formatDayTitle: 'formatDayTitle',
    formatMonthTitle: 'formatMonthTitle', startingDay: 'startingDay', yearRange: 'yearRange',
    onlyCurrentMonth: 'onlyCurrentMonth', shortcutPropagation: 'shortcutPropagation',
    monthColLimit: 'monthColLimit', yearColLimit: 'yearColLimit', customClass: 'customClass',
    dateDisabled: 'dateDisabled', activeDate: 'activeDate' }, { selectionDone: 'selectionDone',
    activeDateChange: 'activeDateChange' }, []);
