// ngx-bootstrap/datepicker/daypicker.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ngx-bootstrap/datepicker/daypicker.component';
import * as i3 from 'ngx-bootstrap/datepicker/datepicker-inner.component';
var styles_DayPickerComponent = ['[_nghost-%COMP%]   .btn-secondary[_ngcontent-%COMP%] {\n      color: #292b2c;\n      background-color: #fff;\n      border-color: #ccc;\n    }\n    [_nghost-%COMP%]   .btn-info[_ngcontent-%COMP%]   .text-muted[_ngcontent-%COMP%] {\n      color: #292b2c !important;\n    }'];
export var RenderType_DayPickerComponent = i0.ɵcrt({ encapsulation: 0,
    styles: styles_DayPickerComponent, data: {} });
function View_DayPickerComponent_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 3, 'button', [['class',
                'btn btn-default btn-secondary btn-sm pull-left float-left'], ['tabindex', '-1'],
            ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.move((0 - 1)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n          '])),
        (_l()(), i0.ɵeld(2, 0, null, null, 0, 'i', [['class', 'glyphicon glyphicon-chevron-left']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n        ']))], null, null);
}
function View_DayPickerComponent_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'button', [['class',
                'btn btn-default btn-secondary btn-sm pull-left float-left'], ['tabindex', '-1'],
            ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.move((0 - 1)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['<\n        ']))], null, null);
}
function View_DayPickerComponent_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 3, 'button', [['class',
                'btn btn-default btn-secondary btn-sm pull-right float-right'], ['tabindex', '-1'],
            ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.move(1) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n          '])),
        (_l()(), i0.ɵeld(2, 0, null, null, 0, 'i', [['class', 'glyphicon glyphicon-chevron-right']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n        ']))], null, null);
}
function View_DayPickerComponent_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'button', [['class',
                'btn btn-default btn-secondary btn-sm pull-right float-right'], ['tabindex', '-1'],
            ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.move(1) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['>\n        ']))], null, null);
}
function View_DayPickerComponent_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 0, 'th', [], null, null, null, null, null))], null, null);
}
function View_DayPickerComponent_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 5, 'th', [['class',
                'text-center']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵeld(2, 0, null, null, 2, 'small', [['aria-label', 'labelz.full']], null, null, null, null, null)), (_l()(), i0.ɵeld(3, 0, null, null, 1, 'b', [], null, null, null, null, null)), (_l()(), i0.ɵted(4, null, ['', ''])),
        (_l()(), i0.ɵted(-1, null, ['\n      ']))], null, function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.abbr;
        _ck(_v, 4, 0, currVal_0);
    });
}
function View_DayPickerComponent_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'td', [['class',
                'text-center']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n          '])), (_l()(), i0.ɵeld(2, 0, null, null, 1, 'em', [], null, null, null, null, null)), (_l()(), i0.ɵted(3, null, ['', ''])),
        (_l()(), i0.ɵted(-1, null, ['\n        ']))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.weekNumbers[_v.parent.parent.context.index];
        _ck(_v, 3, 0, currVal_0);
    });
}
function View_DayPickerComponent_12(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 8, 'button', [['style',
                'min-width:100%;'], ['tabindex', '-1'], ['type', 'button']], [[8, 'disabled', 0]], [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.select(_v.parent.context.$implicit.date) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(1, 278528, null, 0, i1.NgClass, [i0.IterableDiffers,
            i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0, 'klass'], ngClass: [1,
                'ngClass'] }, null), i0.ɵpod(2, { 'btn-secondary': 0, 'btn-info': 1, disabled: 2,
            active: 3, 'btn-default': 4 }), (_l()(), i0.ɵted(-1, null, ['\n            '])),
        (_l()(), i0.ɵeld(4, 0, null, null, 3, 'span', [], null, null, null, null, null)), i0.ɵdid(5, 278528, null, 0, i1.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngClass: [0, 'ngClass'] }, null), i0.ɵpod(6, { 'text-muted': 0,
            'text-info': 1 }), (_l()(), i0.ɵted(7, null, ['', ''])), (_l()(), i0.ɵted(-1, null, ['\n          ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = i0.ɵinlineInterpolate(1, 'btn btn-sm ', _v.parent.context.$implicit.customClass, '');
        var currVal_2 = _ck(_v, 2, 0, ((_co.isBs4 && !_v.parent.context.$implicit.selected) && !_co.datePicker.isActive(_v.parent.context.$implicit)), _v.parent.context.$implicit.selected, _v.parent.context.$implicit.disabled, (!_co.isBs4 && _co.datePicker.isActive(_v.parent.context.$implicit)), !_co.isBs4);
        _ck(_v, 1, 0, currVal_1, currVal_2);
        var currVal_3 = _ck(_v, 6, 0, (_v.parent.context.$implicit.secondary || _v.parent.context.$implicit.current), (!_co.isBs4 && _v.parent.context.$implicit.current));
        _ck(_v, 5, 0, currVal_3);
    }, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit.disabled;
        _ck(_v, 0, 0, currVal_0);
        var currVal_4 = _v.parent.context.$implicit.label;
        _ck(_v, 7, 0, currVal_4);
    });
}
function View_DayPickerComponent_11(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'td', [['class',
                'text-center'], ['role', 'gridcell']], [[8, 'id', 0]], null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n          '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_12)),
        i0.ɵdid(3, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n        ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = !(_co.datePicker.onlyCurrentMonth && _v.context.$implicit.secondary);
        _ck(_v, 3, 0, currVal_1);
    }, function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.uid;
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_DayPickerComponent_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 7, 'tr', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_10)), i0.ɵdid(3, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_11)), i0.ɵdid(6, 802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n      ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.datePicker.showWeeks;
        _ck(_v, 3, 0, currVal_0);
        var currVal_1 = _v.parent.context.$implicit;
        _ck(_v, 6, 0, currVal_1);
    }, null);
}
function View_DayPickerComponent_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_9)), i0.ɵdid(2, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = !((_co.datePicker.onlyCurrentMonth && _v.context.$implicit[0].secondary) && _v.context.$implicit[6].secondary);
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
function View_DayPickerComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 51, 'table', [['aria-activedescendant',
                'activeDateId'], ['role', 'grid']], [[1, 'aria-labelledby', 0]], null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(),
            i0.ɵeld(2, 0, null, null, 42, 'thead', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵeld(4, 0, null, null, 30, 'tr', [], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(6, 0, null, null, 7, 'th', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_2)),
        i0.ɵdid(9, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n        '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_3)),
        i0.ɵdid(12, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n      '])),
        (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(15, 0, null, null, 9, 'th', [], [[1, 'colspan', 0]], null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])),
        (_l()(), i0.ɵeld(17, 0, null, null, 6, 'button', [['class', 'btn btn-default btn-secondary btn-sm'],
            ['style', 'width:100%;'], ['tabindex', '-1'], ['type', 'button']], [[8, 'id', 0],
            [8, 'disabled', 0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.toggleMode() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(18, 278528, null, 0, i1.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod(19, { disabled: 0 }),
        (_l()(), i0.ɵted(-1, null, ['\n          '])), (_l()(), i0.ɵeld(21, 0, null, null, 1, 'strong', [], null, null, null, null, null)), (_l()(), i0.ɵted(22, null, ['', ''])),
        (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(26, 0, null, null, 7, 'th', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_4)), i0.ɵdid(29, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_5)), i0.ɵdid(32, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵted(-1, null, ['\n    '])),
        (_l()(), i0.ɵeld(36, 0, null, null, 7, 'tr', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_6)), i0.ɵdid(39, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null),
        (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_7)), i0.ɵdid(42, 802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(), i0.ɵeld(46, 0, null, null, 4, 'tbody', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_8)),
        i0.ɵdid(49, 802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(), i0.ɵted(-1, null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = !_co.isBs4;
        _ck(_v, 9, 0, currVal_1);
        var currVal_2 = _co.isBs4;
        _ck(_v, 12, 0, currVal_2);
        var currVal_6 = 'btn btn-default btn-secondary btn-sm';
        var currVal_7 = _ck(_v, 19, 0, (_co.datePicker.datepickerMode === _co.datePicker.maxMode));
        _ck(_v, 18, 0, currVal_6, currVal_7);
        var currVal_9 = !_co.isBs4;
        _ck(_v, 29, 0, currVal_9);
        var currVal_10 = _co.isBs4;
        _ck(_v, 32, 0, currVal_10);
        var currVal_11 = _co.datePicker.showWeeks;
        _ck(_v, 39, 0, currVal_11);
        var currVal_12 = _co.labels;
        _ck(_v, 42, 0, currVal_12);
        var currVal_13 = _co.rows;
        _ck(_v, 49, 0, currVal_13);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.datePicker.uniqueId + '-title');
        _ck(_v, 0, 0, currVal_0);
        var currVal_3 = (5 + (_co.datePicker.showWeeks ? 1 : 0));
        _ck(_v, 15, 0, currVal_3);
        var currVal_4 = (_co.datePicker.uniqueId + '-title');
        var currVal_5 = (_co.datePicker.datepickerMode === _co.datePicker.maxMode);
        _ck(_v, 17, 0, currVal_4, currVal_5);
        var currVal_8 = _co.title;
        _ck(_v, 22, 0, currVal_8);
    });
}
export function View_DayPickerComponent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_DayPickerComponent_1)), i0.ɵdid(2, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.datePicker.datepickerMode === 'day');
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
export function View_DayPickerComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'daypicker', [], null, null, null, View_DayPickerComponent_0, RenderType_DayPickerComponent)), i0.ɵdid(1, 114688, null, 0, i2.DayPickerComponent, [i3.DatePickerInnerComponent], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
export var DayPickerComponentNgFactory = i0.ɵccf('daypicker', i2.DayPickerComponent, View_DayPickerComponent_Host_0, {}, {}, []);
