// ngx-bootstrap/datepicker/yearpicker.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ngx-bootstrap/datepicker/yearpicker.component';
import * as i3 from 'ngx-bootstrap/datepicker/datepicker-inner.component';
var styles_YearPickerComponent = ['[_nghost-%COMP%]   .btn-info[_ngcontent-%COMP%]   .text-success[_ngcontent-%COMP%] {\n      color: #fff !important;\n    }'];
export var RenderType_YearPickerComponent = i0.ɵcrt({ encapsulation: 0,
    styles: styles_YearPickerComponent, data: {} });
function View_YearPickerComponent_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 11, 'td', [['class',
                'text-center'], ['role', 'gridcell']], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])),
        (_l()(), i0.ɵeld(2, 0, null, null, 8, 'button', [['class', 'btn btn-default'],
            ['style', 'min-width:100%;'], ['tabindex', '-1'], ['type', 'button']], [[8, 'disabled',
                0]], [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.select(_v.context.$implicit.date) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(3, 278528, null, 0, i1.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod(4, { 'btn-link': 0,
            'btn-info': 1, disabled: 2, active: 3 }), (_l()(), i0.ɵted(-1, null, ['\n          '])),
        (_l()(), i0.ɵeld(6, 0, null, null, 3, 'span', [], null, null, null, null, null)), i0.ɵdid(7, 278528, null, 0, i1.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef,
            i0.Renderer], { ngClass: [0, 'ngClass'] }, null), i0.ɵpod(8, { 'text-success': 0,
            'text-info': 1 }), (_l()(), i0.ɵted(9, null, ['', ''])), (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵted(-1, null, ['\n      ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = 'btn btn-default';
        var currVal_2 = _ck(_v, 4, 0, ((_co.isBs4 && !_v.context.$implicit.selected) && !_co.datePicker.isActive(_v.context.$implicit)), (_v.context.$implicit.selected || ((_co.isBs4 && !_v.context.$implicit.selected) && _co.datePicker.isActive(_v.context.$implicit))), _v.context.$implicit.disabled, (!_co.isBs4 && _co.datePicker.isActive(_v.context.$implicit)));
        _ck(_v, 3, 0, currVal_1, currVal_2);
        var currVal_3 = _ck(_v, 8, 0, (_co.isBs4 && _v.context.$implicit.current), (!_co.isBs4 && _v.context.$implicit.current));
        _ck(_v, 7, 0, currVal_3);
    }, function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.disabled;
        _ck(_v, 2, 0, currVal_0);
        var currVal_4 = _v.context.$implicit.label;
        _ck(_v, 9, 0, currVal_4);
    });
}
function View_YearPickerComponent_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 4, 'tr', [], null, null, null, null, null)), (_l()(),
            i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_YearPickerComponent_3)), i0.ɵdid(3, 802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n    ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit;
        _ck(_v, 3, 0, currVal_0);
    }, null);
}
function View_YearPickerComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 40, 'table', [['role',
                'grid']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(), i0.ɵeld(2, 0, null, null, 31, 'thead', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n    '])),
        (_l()(), i0.ɵeld(4, 0, null, null, 28, 'tr', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(6, 0, null, null, 6, 'th', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(),
            i0.ɵeld(8, 0, null, null, 3, 'button', [['class', 'btn btn-default btn-sm pull-left float-left'],
                ['tabindex', '-1'], ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
                var ad = true;
                var _co = _v.component;
                if (('click' === en)) {
                    var pd_0 = (_co.datePicker.move((0 - 1)) !== false);
                    ad = (pd_0 && ad);
                }
                return ad;
            }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n          '])),
        (_l()(), i0.ɵeld(10, 0, null, null, 0, 'i', [['class', 'glyphicon glyphicon-chevron-left']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(14, 0, null, null, 9, 'th', [], [[1, 'colspan', 0]], null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵeld(16, 0, null, null, 6, 'button', [['class', 'btn btn-default btn-sm'], ['role', 'heading'], ['style', 'width:100%;'],
            ['tabindex', '-1'], ['type', 'button']], [[8, 'id', 0], [8, 'disabled', 0]], [[null,
                'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.toggleMode() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), i0.ɵdid(17, 278528, null, 0, i1.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer], { klass: [0,
                'klass'], ngClass: [1, 'ngClass'] }, null), i0.ɵpod(18, { disabled: 0 }),
        (_l()(), i0.ɵted(-1, null, ['\n          '])), (_l()(), i0.ɵeld(20, 0, null, null, 1, 'strong', [], null, null, null, null, null)), (_l()(), i0.ɵted(21, null, ['', ''])),
        (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(25, 0, null, null, 6, 'th', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵeld(27, 0, null, null, 3, 'button', [['class', 'btn btn-default btn-sm pull-right float-right'], ['tabindex', '-1'],
            ['type', 'button']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.datePicker.move(1) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n          '])),
        (_l()(), i0.ɵeld(29, 0, null, null, 0, 'i', [['class', 'glyphicon glyphicon-chevron-right']], null, null, null, null, null)),
        (_l()(), i0.ɵted(-1, null, ['\n        '])), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(),
            i0.ɵeld(35, 0, null, null, 4, 'tbody', [], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_YearPickerComponent_2)), i0.ɵdid(38, 802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0,
                'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n  '])),
        (_l()(), i0.ɵted(-1, null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_3 = 'btn btn-default btn-sm';
        var currVal_4 = _ck(_v, 18, 0, (_co.datePicker.datepickerMode === _co.datePicker.maxMode));
        _ck(_v, 17, 0, currVal_3, currVal_4);
        var currVal_6 = _co.rows;
        _ck(_v, 38, 0, currVal_6);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (((_co.datePicker.yearColLimit - 2) <= 0) ? 1 : (_co.datePicker.yearColLimit - 2));
        _ck(_v, 14, 0, currVal_0);
        var currVal_1 = (_co.datePicker.uniqueId + '-title');
        var currVal_2 = (_co.datePicker.datepickerMode === _co.datePicker.maxMode);
        _ck(_v, 16, 0, currVal_1, currVal_2);
        var currVal_5 = _co.title;
        _ck(_v, 21, 0, currVal_5);
    });
}
export function View_YearPickerComponent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_YearPickerComponent_1)), i0.ɵdid(2, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0,
                'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.datePicker.datepickerMode === 'year');
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
export function View_YearPickerComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'yearpicker', [], null, null, null, View_YearPickerComponent_0, RenderType_YearPickerComponent)), i0.ɵdid(1, 114688, null, 0, i2.YearPickerComponent, [i3.DatePickerInnerComponent], null, null)], function (_ck, _v) {
        _ck(_v, 1, 0);
    }, null);
}
export var YearPickerComponentNgFactory = i0.ɵccf('yearpicker', i2.YearPickerComponent, View_YearPickerComponent_Host_0, {}, {}, []);
