// ngx-bootstrap/typeahead/typeahead-container.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from 'ngx-bootstrap/typeahead/typeahead-container.component';
var styles_TypeaheadContainerComponent = [];
export var RenderType_TypeaheadContainerComponent = i0.ɵcrt({ encapsulation: 2,
    styles: styles_TypeaheadContainerComponent, data: {} });
function View_TypeaheadContainerComponent_1(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_TypeaheadContainerComponent_2(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 0, 'span', [], [[8, 'innerHTML', 1]], null, null, null, null))], null, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.hightlight(_v.context.match, _v.context.query);
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_TypeaheadContainerComponent_5(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'li', [['class',
                'dropdown-header']], null, null, null, null, null)), (_l()(), i0.ɵted(1, null, ['', '']))], null, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_TypeaheadContainerComponent_7(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_TypeaheadContainerComponent_6(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 8, 'li', [], [[2, 'active', null]], [[null, 'mouseenter']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('mouseenter' === en)) {
                var pd_0 = (_co.selectActive(_v.parent.context.$implicit) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n      '])),
        (_l()(), i0.ɵeld(2, 0, null, null, 5, 'a', [['href', '#'], ['tabindex',
                '-1']], null, [[null, 'click']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.selectMatch(_v.parent.context.$implicit, $event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n        '])),
        (_l()(), i0.ɵand(16777216, null, null, 2, null, View_TypeaheadContainerComponent_7)),
        i0.ɵdid(5, 540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'], ngOutletContext: [1, 'ngOutletContext'] }, null),
        i0.ɵpod(6, { item: 0, index: 1, match: 2, query: 3 }), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵted(-1, null, ['\n    ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = (_co.itemTemplate || i0.ɵnov(_v.parent.parent.parent, 7));
        var currVal_2 = _ck(_v, 6, 0, _v.parent.context.$implicit.item, _v.parent.context.index, _v.parent.context.$implicit, _co.query);
        _ck(_v, 5, 0, currVal_1, currVal_2);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.isActive(_v.parent.context.$implicit);
        _ck(_v, 0, 0, currVal_0);
    });
}
function View_TypeaheadContainerComponent_4(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n    '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TypeaheadContainerComponent_5)),
        i0.ɵdid(2, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n    '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TypeaheadContainerComponent_6)),
        i0.ɵdid(5, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n  ']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.isHeader();
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.isHeader();
        _ck(_v, 5, 0, currVal_1);
    }, null);
}
function View_TypeaheadContainerComponent_3(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵeld(1, 0, null, null, 4, 'ul', [['class', 'dropdown-menu']], null, null, null, null, null)), (_l()(), i0.ɵted(-1, null, ['\n  '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TypeaheadContainerComponent_4)), i0.ɵdid(4, 802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef, i0.IterableDiffers], { ngForOf: [0,
                'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(),
            i0.ɵted(-1, null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.matches;
        _ck(_v, 4, 0, currVal_0);
    }, null);
}
function View_TypeaheadContainerComponent_10(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'h6', [['class',
                'dropdown-header']], null, null, null, null, null)), (_l()(), i0.ɵted(1, null, ['', '']))], null, function (_ck, _v) {
        var currVal_0 = _v.parent.context.$implicit;
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_TypeaheadContainerComponent_12(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵand(0, null, null, 0))], null, null);
}
function View_TypeaheadContainerComponent_11(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵeld(1, 0, null, null, 5, 'button', [['class', 'dropdown-item']], [[2, 'active',
                null]], [[null, 'click'], [null, 'mouseenter']], function (_v, en, $event) {
            var ad = true;
            var _co = _v.component;
            if (('click' === en)) {
                var pd_0 = (_co.selectMatch(_v.parent.context.$implicit, $event) !== false);
                ad = (pd_0 && ad);
            }
            if (('mouseenter' === en)) {
                var pd_1 = (_co.selectActive(_v.parent.context.$implicit) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)), (_l()(), i0.ɵted(-1, null, ['\n          '])),
        (_l()(), i0.ɵand(16777216, null, null, 2, null, View_TypeaheadContainerComponent_12)),
        i0.ɵdid(4, 540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0,
                'ngTemplateOutlet'], ngOutletContext: [1, 'ngOutletContext'] }, null),
        i0.ɵpod(5, { item: 0, index: 1, match: 2, query: 3 }), (_l()(), i0.ɵted(-1, null, ['\n      '])), (_l()(), i0.ɵted(-1, null, ['\n  ']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_1 = (_co.itemTemplate || i0.ɵnov(_v.parent.parent.parent, 7));
        var currVal_2 = _ck(_v, 5, 0, _v.parent.context.$implicit.item, _v.parent.context.index, _v.parent.context.$implicit, _co.query);
        _ck(_v, 4, 0, currVal_1, currVal_2);
    }, function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.isActive(_v.parent.context.$implicit);
        _ck(_v, 1, 0, currVal_0);
    });
}
function View_TypeaheadContainerComponent_9(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n   '])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TypeaheadContainerComponent_10)),
        i0.ɵdid(2, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n   '])),
        (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TypeaheadContainerComponent_11)),
        i0.ɵdid(5, 16384, null, 0, i1.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, 'ngIf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n']))], function (_ck, _v) {
        var currVal_0 = _v.context.$implicit.isHeader();
        _ck(_v, 2, 0, currVal_0);
        var currVal_1 = !_v.context.$implicit.isHeader();
        _ck(_v, 5, 0, currVal_1);
    }, null);
}
function View_TypeaheadContainerComponent_8(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_TypeaheadContainerComponent_9)),
        i0.ɵdid(2, 802816, null, 0, i1.NgForOf, [i0.ViewContainerRef, i0.TemplateRef,
            i0.IterableDiffers], { ngForOf: [0, 'ngForOf'] }, null), (_l()(), i0.ɵted(-1, null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = _co.matches;
        _ck(_v, 2, 0, currVal_0);
    }, null);
}
export function View_TypeaheadContainerComponent_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵand(16777216, null, null, 2, null, View_TypeaheadContainerComponent_1)), i0.ɵdid(3, 540672, null, 0, i1.NgTemplateOutlet, [i0.ViewContainerRef], { ngTemplateOutlet: [0, 'ngTemplateOutlet'], ngOutletContext: [1,
                'ngOutletContext'] }, null), i0.ɵpod(4, { matches: 0, itemTemplate: 1, query: 2 }),
        (_l()(), i0.ɵted(-1, null, ['\n\n'])), (_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵand(0, [['bsItemTemplate', 2]], null, 0, null, View_TypeaheadContainerComponent_2)), (_l()(), i0.ɵted(-1, null, ['\n\n'])),
        (_l()(), i0.ɵted(-1, null, ['\n'])), (_l()(), i0.ɵand(0, [['bs3Template',
                2]], null, 0, null, View_TypeaheadContainerComponent_3)), (_l()(),
            i0.ɵted(-1, null, ['\n\n'])), (_l()(), i0.ɵted(-1, null, ['\n'])),
        (_l()(), i0.ɵand(0, [['bs4Template', 2]], null, 0, null, View_TypeaheadContainerComponent_8)),
        (_l()(), i0.ɵted(-1, null, ['\n']))], function (_ck, _v) {
        var _co = _v.component;
        var currVal_0 = (_co.optionsListTemplate || (_co.isBs4 ? i0.ɵnov(_v, 13) : i0.ɵnov(_v, 10)));
        var currVal_1 = _ck(_v, 4, 0, _co.matches, _co.itemTemplate, _co.query);
        _ck(_v, 3, 0, currVal_0, currVal_1);
    }, null);
}
export function View_TypeaheadContainerComponent_Host_0(_l) {
    return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, 'typeahead-container', [['class', 'dropdown open'], ['style', 'position: absolute;display: block;']], [[2,
                'dropdown-menu', null], [2, 'dropup', null]], [[null,
                'mouseleave'], [null, 'blur']], function (_v, en, $event) {
            var ad = true;
            if (('mouseleave' === en)) {
                var pd_0 = (i0.ɵnov(_v, 1).focusLost() !== false);
                ad = (pd_0 && ad);
            }
            if (('blur' === en)) {
                var pd_1 = (i0.ɵnov(_v, 1).focusLost() !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, View_TypeaheadContainerComponent_0, RenderType_TypeaheadContainerComponent)),
        i0.ɵdid(1, 49152, null, 0, i2.TypeaheadContainerComponent, [i0.ElementRef], null, null)], null, function (_ck, _v) {
        var currVal_0 = i0.ɵnov(_v, 1).isBs4;
        var currVal_1 = i0.ɵnov(_v, 1).dropup;
        _ck(_v, 0, 0, currVal_0, currVal_1);
    });
}
export var TypeaheadContainerComponentNgFactory = i0.ɵccf('typeahead-container', i2.TypeaheadContainerComponent, View_TypeaheadContainerComponent_Host_0, {}, {}, []);
